src/Plugin/AddPathPlugin.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Http\Client\Common\Plugin;
+
+use Http\Client\Common\Plugin;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Add base path to the request URI. Useful for base API URLs like http://domain.com/api.
+ *
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+final class AddPathPlugin implements Plugin
+{
+    /**
+     * @var UriInterface
+     */
+    private $host;
+
+    /**
+     * @param UriInterface $host
+     */
+    public function __construct(UriInterface $host)
+    {
+        if (rtrim($host->getPath(), '/') === '') {
```

Comment:
i think i would check '' or last character is '/' and throw an exception when it is. silently fixing things makes harder debugging. after doing example.com/ once or twice, people will learn to use example.com